src/TuDa.CIMS.Web/Services/IPurchaseApi.cs

```diff
@@ -0,0 +1,127 @@
+
+    /// <summary>
+    /// Internal method to create a new purchase.
+    /// </summary>
+    /// <param name="item">The JSON serialized representation of the purchase.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="IApiResponse{T}"/> containing the unique identifier of the created purchase.
+    /// </returns>
+    /// <remarks>This will be generated by <see cref="Refit"/>.</remarks>
+    [Post("/")]
+    [Headers("Content-Type: application/json; charset=utf-8")]
+    protected Task<IApiResponse<Guid>> CreateAsyncInternal([Body] string item);
+
+    /// <summary>
+    /// Updates an existing purchase.
+    /// </summary>
+    /// <param name="id">The unique identifier of the purchase to update.</param>
+    /// <param name="updatePurchaseDto">The updatePurchaseDto.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="ErrorOr{T}"/> containing a status indicating the update success.
+    /// </returns>
+    /// <remarks>
+    /// Uses <see cref="JsonSerializer"/> to ensure proper serialization of polymorphic types.
+    /// </remarks>
+    public async Task<ErrorOr<Updated>> UpdateAsync(Guid id, UpdatePurchaseDto updatePurchaseDto) =>
+        await UpdateAsyncInternal(id, JsonSerializer.Serialize(updatePurchaseDto)).ToErrorOrAsync();
+
+    /// <summary>
+    /// Internal method to update an existing purchase.
+    /// </summary>
+    /// <param name="id">The unique identifier of the purchase to update.</param>
+    /// <param name="updatePurchaseDto">The JSON serialized representation of the updatedPurchaseDto.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="IApiResponse"/>.
+    /// </returns>
+    /// <remarks>This will be generated by <see cref="Refit"/>.</remarks>
+    [Patch("/{id}")]
+    protected Task<IApiResponse<Updated>> UpdateAsyncInternal(Guid id, [Body] string updatePurchaseDto);
+
+    /// <summary>
+    /// Removes a purchase by its unique identifier.
+    /// </summary>
+    /// <param name="id">The unique identifier of the purchase to remove.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="ErrorOr{T}"/> containing a status indicating the deletion.
+    /// </returns>
+    public async Task<ErrorOr<Deleted>> RemoveAsync(Guid id) => await RemoveAsyncInternal(id).ToErrorOrAsync<Deleted>();
+
+    /// <summary>
+    /// Internal method to remove a purchase by its unique identifier.
+    /// </summary>
+    /// <param name="id">The unique identifier of the purchase to remove.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="IApiResponse"/>.
+    /// </returns>
+    /// <remarks>This will be generated by <see cref="Refit"/>.</remarks>
+    [Delete("/{id}")]
+    [Headers("Content-Type: application/json; charset=utf-8")]
```

Comment:
🔧 This is not needed here. It's only needed when serializing an object manually, to ensure the content type